[e2e] added repetitions to ensure the state of the cluster

adrianriobo opened the pull request:
During e2e executions eventually we see executions when the cluster state is showing running and then is degraded right away, if the subsequent steps on an scneario depends on that state we need to ensure the stability of the cluster. This PR introduces a new param `repetitions` to define the number of times that the state of the cluster should keep stable for a period of time

praveenkumar commented on test/extended/util/util.go:
```diff
@@ -144,41 +143,34 @@ func RemoveCRCHome(crcHome string) error {
-		err := match(expression)
-		if err == nil {
-			return nil
+// MatchWithRetry will execute match function with expression as arg
```
it should be `MatchRepetitionsWithRetry` instead `MatchWitRetry` for this function.

adrianriobo replied:
done

adrianriobo committed:
[e2e] added repetitions for MatchWithRetry

this ensures certain level of stability on the cluster state; now checking the state of the cluster the state should be matched a number of times in a row during the amount of time. This will remove states when for a small period the cluster is running but then is degraded right away

gbraad commented on test/extended/crc/cmd/cmd.go:
```diff
@@ -12,8 +12,10 @@ import (
-	clusterStateTimeout       = 600
+	clusterStateRetryCount = 25
+	clusterStateTimeout    = 600
+	// defines the number of times the state should be matches in a row
```
matched

praveenkumar commented on test/extended/util/util.go:
```diff
@@ -144,41 +143,34 @@ func RemoveCRCHome(crcHome string) error {
+		case <-tick.C:
+			if err := match(expression); err == nil {
+				matchRepetition++
+				if matchRepetition == matchRepetitions {
```
@adrianriobo so with this logic we want to make sure atleast 3 time in a row the cluster state should be stable?

adrianriobo replied:
yeah...basically the point is ..we want to ensure the cluster is healthy in less than 20 minutes...but healthy does not mean one hit on `state=running`. As I found cases where tests failed b/c cluster state was running and right away went to starting again...

so e2e step (checking the state of cluster was passed) and the next one...lets say `oc login` failed... 

So to prevent to situation...and based on the logic from `crc start` I wanna ensure the health of the cluster based on this 3 in a row running state checks
